main/src/com/google/refine/importing/EncodingGuesser.java

```diff
@@ -38,9 +39,13 @@ public final static void guess(final ImportingJob job)
                     if (StringUtils.isBlank(encoding)) {
                         String location = JSONUtilities.getString(record, "location", null);
                         if (location != null) {
-                            try (InputStream is = new BufferedInputStream(
+                            try (UnicodeBOMInputStream is = new UnicodeBOMInputStream(
                                     new FileInputStream(new File(job.getRawDataDir(), location)))) {
                                 String detected = UniversalDetector.detectCharset(is);
+                                UnicodeBOMInputStream.BOM bom = is.getBOM();
+                                if (UnicodeBOMInputStream.BOM.UTF_8.equals(bom)) {
+                                    detected = UTF_8_BOM;
+                                }
```

Comment:
There are other types of BOM markers supported (such as for UTF-16, UTF-32, with different byte orders) - would it make sense to add support for those as well?

Reply:
Are other BOM types problematic? I've never heard any complaints about them. Given that this bug report has been open for 5 years, I'd lean towards addressing the specific problem that was reported rather than delaying for additional investigation (unless you know what needs to be adjusted).

Reply:
Ah, I had not understood that there was a specific issue in the JDK about UTF-8 and BOM handling. According to https://github.com/gpakosz/UnicodeBOMInputStream#why, it seems that UTF-16 and UTF-32 are indeed handled correctly out of the box.

Reply:
Actually, the issue is with Microsoft applications inserting a BOM with UTF-8 in a context where it is "[neither required nor recommended](https://en.wikipedia.org/wiki/Byte_order_mark#cite_note-5)"

Where does this PR stand with regards to approval status?